src/modes/direct_select.js

```diff
@@ -73,6 +73,12 @@ DirectSelect.onMidpoint = function (state, e) {
   state.feature.addCoordinate(about.coord_path, about.lng, about.lat);
   this.fireUpdate();
   state.selectedCoordPaths = [about.coord_path];
+
+  const selectedCoordinates = this.pathsToCoordinates(
+    state.featureId,
+    state.selectedCoordPaths
+  );
+  this.setSelectedCoordinates(selectedCoordinates);
```

Comment:
cool Im sure this is fine, hows it work though? / what was causing the bug?

Reply:
`this.setSelectedCoordinates` will set the selected coords in `store.js`. if you see line 67, we are doing it for vertex point. because we are missing this function for mid segment point, every time we finish the dragging of mid point, it will use the previous coords stored in `store.js` to do a snapping to the target line. that's why the additional snapping happens.